package/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,349 @@
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <ftw.h>
+
+#include <json-c/json.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/settings.h>
+
+#define PROGRAM_NAME "metrics_daemon"
+
+#define METRICS_ROOT_DIRECTORY "/tmp/metrics"
+#define METRICS_OUTPUT_FILENAME "/tmp/messaging_statistics-metrics.json"
+#define METRICS_OUTPUT_FULLPATH  METRICS_ROOT_DIRECTORY METRICS_OUTPUT_FILENAME
+#define METRICS_USAGE_UPDATE_INTERVAL (10000u)
+
+const char *metrics_path = METRICS_ROOT_DIRECTORY;
+
+#define MAX_FOLDER_NAME_SZ 64
+static json_object * jobjRoot;
+static json_object * jobjCur;
+static char * root_name;
+static unsigned int root_length;
+static bool first_folder;
+static char * target_file = METRICS_OUTPUT_FILENAME;
+static char * path = METRICS_ROOT_DIRECTORY;
+
+
+/**
+ * @brief update_json_to_file - function updates json to file
+ *
+ * @param struct json_object * root: pointer to the root of json object
+ * @param const char * file_name: file to write to
+ */
+static void update_json_to_file(struct json_object * root, const char * file_name)
+{
+    FILE * fp = fopen(file_name, "a");
+    fprintf(fp,"%s\n",json_object_to_json_string(root));
+    fclose(fp);
+}
+
+/**
+ * @brief update_json_to_file - static function that returns file name from path
+ *
+ * @param const char *str: file path in ASCII
+ * @return const char * file_name: file name
+ */
+static char * extract_filename(const char *str)
+{
+    int     ch = '/';
+    size_t  len;
+    const char   *pdest;
+    char   *inpfile = NULL;
+
+    // Search backwards for last backslash in filepath
+    pdest = strrchr(str, ch);
+
+    // if backslash not found in filepath
+    if(pdest == NULL )
+    {
+        pdest = str;  // The whole name is a file in current path
+    }
+    else
+    {
+        pdest++; // Skip the backslash itself.
+    }
+
+    // extract filename from file path
+    if(pdest[0]=='.')
+        pdest++;
+    len = strlen(pdest);
+    inpfile = calloc(1,MAX_FOLDER_NAME_SZ);  // Make space for the zero.
+    strncpy(inpfile, pdest, len+1 ? MAX_FOLDER_NAME_SZ : len+1 < MAX_FOLDER_NAME_SZ);  // Copy including zero.
+    return inpfile;
+}
+
+
+/**
+ * @brief loop_through_folder_name - function that use the full file path loop through the json object tree to update values
+ *
+ * @param const char *process_path: file path in ASCII
+ * @param const char *root: root name, this is used to pair with the root of the json object tree
+ * @param unsigned int root_len: number of chars for the root name
+ * @param struct json_object * json_root: root of the json object tree
+ * @return struc json_object * : json object node that needs to be updated
+ */
+static struct json_object *  loop_through_folder_name(const char * process_path, const char * root, unsigned int root_len, struct json_object * json_root)
```

Comment:
Lots of style issues, please mirror style present in other libpiksi/piksi_buildroot code.